File: SORT_Insertion.cpp
#include <iostream>
#include <string>

int main(){
	
	int min = -1000;
	int max =  1000;
	int inputSize = 10;
	int numOfOperations = 0;
	int arrayToSort[inputSize];

	for(size_t i = 0; i < inputSize; i++){
		arrayToSort[i] = rand()%(max-min + 1) + min;
	}

	std::cout << "Sorting process started ..." << std::endl;
	
	for (int j=0; j<inputSize; j++){
		int key = arrayToSort[j];
		int i = j - 1;
		while(i > 0 & arrayToSort[i] > key){
			arrayToSort[i+1] = arrayToSort[i];
			i -= 1;
			numOfOperations++;
		}
		arrayToSort[i+1] = key;
	}

	std::cout << "Sorting process finished after "<< numOfOperations << " operations."<< std::endl;

	for(size_t i = 0; i < inputSize; i++){
		std::cout << arrayToSort[i];
	}


	return 0;
}

File: SORT_Insertion.py
# ----- Python ----------------------------------------- #
# ----- Sorting Algorithm by Insertion Method ---------- #
# ----- Approximate number of iterations 0.25*n*n ------ #
# ------------------------------------------------------ #
import random

listToSort = []

# Generating a list of random numbers
for i in range(10000):
    listToSort.append( random.randint(-10000, 10000))

inputSize = len(listToSort)
numOfIterations = 0

print(listToSort)

print('Sorting started')

for j in range(1, inputSize, 1):
    key = listToSort[j]
    i = j-1
    while ( key < listToSort[i] and i >= 0):
        listToSort[i+1] = listToSort[i]
        i -= 1
        numOfIterations += 1
    listToSort[i+1] = key

print('Sorting finished after ', numOfIterations, ' iterations.')

print(listToSort)
